PR: https://github.com/odoo/odoo/pull/77081

From: 7bfb6d7f01dc6c598b7a5951f6fe87e21add2cfa
From: Patrick Hoste
Date: 2021-10-04 15:30:26

Structural Changes: 4.8
Total Changes: 11

[FIX] (website_)event_booth(_sale): fix display default booth category image

PURPOSE

Before this commit when a booth category doesn't have an image a
traceback is shown when accessing to the /booths route.
After this commit a default image is shown when the booth category
doesn't have an image.

SPECIFICATION

To allow the public user to see the booth category image we give
everybody the read permission on the model but restrict some
sensitive fields.

LINKS

Task-2635461

Pr: odoo/odoo#77081
Related: odoo/upgrade#2869
Signed-off-by: awa-odoo <awa-odoo@users.noreply.github.com>

================================= pseudo patch: =================================

--- a/addons/event_booth_sale/models/event_booth_category.py
+++ b/addons/event_booth_sale/models/event_booth_category.py
@@ -16,12 +16,15 @@ class EventBoothCategory(models.Model):
 
     product_id = fields.Many2one(
         'product.product', string='Product', required=True,
-        domain=[('detailed_type', '=', 'event_booth')], default=_default_product_id)
-    price = fields.Float(string='Price', compute='_compute_price', digits='Product Price', readonly=False, store=True)
-    currency_id = fields.Many2one(related='product_id.currency_id')
+        domain=[('detailed_type', '=', 'event_booth')], default=_default_product_id,
+        groups="event.group_event_registration_desk")
+    price = fields.Float(
+        string='Price', compute='_compute_price', digits='Product Price', readonly=False,
+        store=True, groups="event.group_event_registration_desk")
+    currency_id = fields.Many2one(related='product_id.currency_id', groups="event.group_event_registration_desk")
     price_reduce = fields.Float(
         string='Price Reduce', compute='_compute_price_reduce',
-        compute_sudo=True, digits='Product Price')
+        compute_sudo=True, digits='Product Price', groups="event.group_event_registration_desk")
     image_1920 = fields.Image(compute='_compute_image_1920', readonly=False, store=True)
 
     @api.depends('product_id')
